oauthproxy.go

```diff
@@ -318,7 +318,12 @@ func (p *OAuthProxy) redeemCode(host, code string) (s *providers.SessionState, e
 		return nil, errors.New("missing code")
 	}
 	redirectURI := p.GetRedirectURI(host)
-	s, err = p.provider.Redeem(redirectURI, code)
+
+	redeemURL := p.provider.GetRedeemURL()
+	if redeemURL == nil {
+		return s, fmt.Errorf("unable to discover the redeeem endpoint")
+	}
+	s, err = p.provider.Redeem(redeemURL, redirectURI, code)
```

Comment:
Let's put the reedemURL stuff into the `provider.Redeem` function and keep the function signature the same.

Reply:
I think this is a good suggestion, otherwise the PR looks good.

Reply:
https://github.com/openshift/oauth-proxy/pull/101/files#r256425827 explains why I put the `Get*()` functions outside